lmnet/lmnet/networks/quantize_param_init.py

```diff
@@ -0,0 +1,123 @@
+import tensorflow as tf
+import functools
+
+
+class QuantizeParamInit:
+    """QuantizeParamInit is a mixin creating a quantized network class more convenient by providing two functions.
+    1) __init___ : for initializing variables necessary for quantization, such as quantization function
+    2) _quantized_variable_getter : for quantizing weight variables in convolutional layers
+
+    This QuantizeParamInit class can be used for all kinds of task (classification, object detection, and segmentation).
+
+    How to use
+    1) Extend this mixin
+    2) Set the custom_getter variable to `self.custom_getter` if its base class does not set it yet
+       The setting is done by adding the base function defining custom_getter for tensorflow inside the class as follows
+             def base(self, images, is_training):
+                 with tf.variable_scope("", custom_getter=self.custom_getter):
+                     return super().base(images, is_training)
+
+    In the object construction step, users must pass
+       * arguments for the quantization (e.g. activation_quantizer) followed by
+       * arguments of the subtask class (e.g. weight_decay_rate)
+       to the init function, this mixin will call init function of subtask's base class automatically.
+
+    Requirement
+    1) The base class must define `self.first_layer_name` and `self.last_layer_name` in the init function, where
+       `self.first_layer_name` = prefix of the first convolutional layer and
+       `self.last_layer_name` = prefix of the last convolutional layer
+
+    Args:
+        activation_quantizer (callable): An activation quantization function
+        activation_quantizer_kwargs (dict): A dictionary of arguments for the activation quantization function
+        weight_quantizer (callable): A weight quantization function
+        weight_quantizer_kwargs (dict): A dictionary of arguments for the weight quantization function
+        quantize_first_convolution (boolean): True if using quantization at the first layer, False otherwise
+        quantize_last_convolution (boolean): True if using quantization at the last layer, False otherwise
+        *args: argument for the initialization of subtask's network class
+        **kwargs: keyword arguments for the initialization of subtask's network class
+    """
+
+    def __init__(
+            self,
+            activation_quantizer=None,
+            activation_quantizer_kwargs=None,
+            weight_quantizer=None,
+            weight_quantizer_kwargs=None,
+            quantize_first_convolution=None,
```

Comment:
Why the `quantize_first_convolution` and `quantize_last_convolution` default value is `None` ? Why not boolean ?

Reply:
I don't know what is the appropriate default value. (implement or not implement quantize at first layer).
However, before implementing this PR, the first layer is quantized by default 
so if the `quantize_first_convolution` is `None`, the first layer is quantized. 

After you mentioned, setting the default value to `True` is easier to understand and maintenance so I will set the default value to `True`.